Sources/Build/Destination.swift

```diff
@@ -0,0 +1,146 @@
+import Foundation
+import Basic
+import PackageModel
+
+enum HostDestinationError: Swift.Error {
+    /// Couldn't find the Xcode installation.
+    case invalidInstallation(problem: String)
+}
+
+extension HostDestinationError: CustomStringConvertible {
+    var description: String {
+        switch self {
+        case .invalidInstallation(let problem):
+            return problem
+        }
+    }
+}
+
+enum DestinationFileError: Swift.Error {
+    /// Wrong version.
+    case destinationFileVersionMismatch(problem: String)
+}
+
+extension DestinationFileError: CustomStringConvertible {
+    var description: String {
+        switch self {
+        case .destinationFileVersionMismatch(let problem):
+            return problem
+        }
+    }
+}
+
+public struct Destination {
+    public let target: String
+
+    public let sdk: AbsolutePath
+
+    public let toolchain: AbsolutePath
+
+    public let dynamicLibraryExtension: String
+
+    public let extraCCFlags: [String]
+
+    public let extraSwiftCFlags: [String]
+
+    public let extraCPPFlags: [String]
+
+    public static func hostDestination() throws -> Destination {
+        #if os(macOS)
+            let sdkPath = try Process.checkNonZeroExit(args: "xcrun", "--sdk", "macosx", "--show-sdk-path").chomp()
+            guard !sdkPath.isEmpty else {
+                throw HostDestinationError.invalidInstallation(problem: "could not find default SDK")
+            }
+
+            let platformPath = try Process.checkNonZeroExit(args: "xcrun", "--sdk", sdkPath, "--show-sdk-platform-path").chomp()
+            guard !platformPath.isEmpty else {
+                throw HostDestinationError.invalidInstallation(problem: "could not find SDK platform path for SDK \(sdkPath)")
+            }
+
+            let sdkPlatformFrameworksPath = AbsolutePath(platformPath).appending(components: "Developer", "Library", "Frameworks")
+
+            let devDirPath = try Process.checkNonZeroExit(args: "xcode-select", "-p").chomp()
+            let sysFWPath = AbsolutePath(sdkPath).appending(component: "System")
```

Comment:
Please avoid abbreviated variable names. We try to follow Swift design guidelines https://swift.org/documentation/api-design-guidelines/